### DISCUSSION TITLE
Can't understand an error appears with my model.

### DISCUSSION THREAD

**Discussion by ghost** [prisma/prisma#14938] [Q&A]
### Bug description

My model appears in the studio, but not working when i'm trying to find one of these records.

### How to reproduce

dont

### Expected behavior

Supposed to send a single record from my database. I got an undefined error : 
```error - (api)\bdd\get\getVedette.js (5:26) @ getVedette
TypeError: Cannot read properties of undefined (reading 'findFirst')
  3 | export default async function getVedette() {
  4 |     const prisma = connectBdd()
> 5 |     return prisma.vedette.findFirst()
    |                          ^
  6 |     .then((vedette) => vedette)
  7 |     .catch(() => false)
  8 |     .finally(() => prisma.$disconnect())```
```

### Prisma information

```npm prisma -v.8.14.0```

```model User {
  id                     Int         @id @unique @default(autoincrement())
  email                  String      @unique
  name                   String      @unique
  surname                String      @default("")
  password               String
  role                   Role        @default(READER)
  created_at             DateTime    @default(now())
  lastCnnct              DateTime    @default(now())
  bio                    String?
  age                    Int?
  private                Boolean     @default(false)
  language               Lang        @default(EN)
  website                String?
  imgsrc                 String?
  imgbann                String?
  verified               Boolean
  challenge              Challenge[]
  notes                  Note[]
  posts                  Post[]
  profile                Profile[]
  lib                    Statebook[]
  User_B_UserBlockedUser User[]      @relation("UserBlockedUser")
  User_A_UserBlockedUser User[]      @relation("UserBlockedUser")
  User_B_UserFriendsUser User[]      @relation("UserFriendsUser")
  User_A_UserFriendsUser User[]      @relation("UserFriendsUser")
}

model Post {
  id        Int      @id @unique @default(autoincrement())
  title     String?
  createdAt DateTime @default(now())
  content   String?
  published Boolean  @default(true)
  bookId    Int
  note      Int
  authorId  Int
  author    User     @relation(fields: [authorId], references: [id])
  book      Book     @relation(fields: [bookId], references: [id])
}

model Note {
  id        Int      @id @unique @default(autoincrement())
  createdAt DateTime @default(now())
  bookId    Int
  note      Int
  authorId  Int
  author    User     @relation(fields: [authorId], references: [id])
  book      Book     @relation(fields: [bookId], references: [id])
}

model Profile {
  id       Int       @id @unique @default(autoincrement())
  born     DateTime?
  dead     DateTime?
  authorId Int
  user     User      @relation(fields: [authorId], references: [id])
  written  Book[]
}

model Book {
  id        Int         @id @unique @default(autoincrement())
  title     String
  lang      Lang        @default(EN)
  agefrom   Int?
  ageto     Int?
  nmbrpages Int?
  poids     Int?
  theme     Theme[]
  resume    String
  parution  DateTime    @default(now())
  serie     Boolean
  published Boolean
  tome      Int         @default(1)
  imgsrc    String      @unique
  isbn10    String      @unique
  isbn13    String      @unique
  ean       String      @unique
  sku       String      @unique
  score     Decimal?
  seen      DailyView[]
  vedette   Vedette[]
  note      Note[]
  avis      Post[]
  state     Statebook[]
  challenge Challenge[]
  authors   Profile[]
}

model DailyView {
  id     Int  @id @unique @default(autoincrement())
  count  Int  @default(0)
  bookId Int
  book   Book @relation(fields: [bookId], references: [id])
}

model Statebook {
  id        Int      @id @unique @default(autoincrement())
  userId    Int
  state     State
  bookId    Int
  favorite  Boolean  @default(false)
  setAsRead DateTime
  book      Book     @relation(fields: [bookId], references: [id])
  user      User     @relation(fields: [userId], references: [id])
}

model Vedette {
  id     Int  @id @unique @default(autoincrement())
  idBook Int
  book   Book @relation(fields: [idBook], references: [id])
}

model Challenge {
  id       Int    @id @unique @default(autoincrement())
  objectif Int
  userId   Int
  year     Int
  user     User   @relation(fields: [userId], references: [id])
  books    Book[]
}```

### Environment & setup

- OS: IOS and Windows
- Browser: all
- Database: prisma + heroku postgres


### Prisma logs

```
  1 | error - (api)\bdd\get\getVedette.js (5:26) @ getVedette
  2 | TypeError: Cannot read properties of undefined (reading 'findFirst')
  3 | export default async function getVedette() {
  4 |     const prisma = connectBdd()
197 | }```

**Comment by petradonka** [comment]
@lucasbriandbr, do I understand right that Studio works as expected, but when you're programatically trying to query for the same record that shows up in Studio, you get the `TypeError: Cannot read properties of undefined (reading 'findFirst')` error?

**Comment by ghost** [comment]
@petradonka yes, absolutely ! I can't make this model (the only one who dont works) after prisma migrate dev, prisma db push, prisma generate, prisma studio. And it appears in the studio, i can add records etc...

**Comment by petradonka** [comment]
Sounds like an issue in https://github.com/prisma/prisma, not in Studio, as Studio is working as you'd expect - I'll transfer this issue to prisma/prisma, where maybe @janpio or another team member can help you figure the issue out.

**Comment by janpio** [comment]
The error message here is saying that `prisma.vedette` is `undefined` and hence has no `firstFirst`.

```
const prisma = connectBdd()
return prisma.vedette.findFirst()
```

What is `connectBdd()` doing? Is it returning a PrismaClient object?

**Reply by ghost** [reply]
absolutely, here is my connectbdd() : 
```
import { PrismaClient } from '@prisma/client'

export default function connectBdd() {
  const prisma = new PrismaClient()
  return prisma
}

**Reply by janpio** [reply]
If you don't use the separate function, but just write the code inline - does it work then? Did you `prisma generate` your Prisma Client?

**Reply by vladi-stevanovic** [reply]
👋 Hello @lucasbriandbr I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Reply by Gwitchr** [reply]
just had this error and it took me like an hour to figure that I had updated the prisma client files without restarting dev, so I guess an old version of in the node_modules was being loaded that didn't have the new methods? but they were loading in the ts helper, anyway maybe it helps someone that comes looking for this issue like I did.

**Reply by danielchikaka** [reply]
> just had this error and it took me like an hour to figure that I had updated the prisma client files without restarting dev, so I guess an old version of in the node_modules was being loaded that didn't have the new methods? but they were loading in the ts helper, anyway maybe it helps someone that comes looking for this issue like I did.

Thanks man, you saved my life ;)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
